src/Microsoft.AspNet.Mvc.Core/ActionConstraints/ConsumesAttribute.cs

```diff
@@ -0,0 +1,135 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNet.Mvc.HeaderValueAbstractions;
+
+namespace Microsoft.AspNet.Mvc
+{
+    /// <summary>
+    /// Specifies the allowed content types which can be used to help in selecting the action based on request's
+    /// content-type.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Method, AllowMultiple = false, Inherited = true)]
+    public class ConsumesAttribute : Attribute, IResourceFilter, IConsumesActionConstraint
+    {
+        /// <summary>
+        /// Creates a new instance of <see cref="ConsumesAttribute"/>.
+        /// </summary>
+        public ConsumesAttribute(string contentType, params string[] otherContentTypes)
+        {
+            ContentTypes = GetContentTypes(contentType, otherContentTypes);
+        }
+
+        public int Order { get; } = Int32.MaxValue - 10;
+
+        public IList<MediaTypeHeaderValue> ContentTypes { get; set; }
+
+        public void OnResourceExecuting([NotNull] ResourceExecutingContext context)
+        {
+            MediaTypeHeaderValue requestContentType = null;
+            MediaTypeHeaderValue.TryParse(context.HttpContext.Request.ContentType, out requestContentType);
+            if (context.ActionDescriptor.FilterDescriptors.Last().Filter == this)
```

Comment:
this check isn't right. You want to execute only if you are the last `consumes filter`. Adding any unrelated filter will break this.